### PR TITLE
avoid panic while encrypting empty data

### LINKED ISSUE #95: EncryptUpdate panics on empty input
I am using the openssl library to encrypt/decrypt data.

I have a Unit test to encrypt/decrypt empty data ( like: `[]byte("")` ). It causes a panic as follows:

```
--- FAIL: TestEmptyEncryption (0.00s)
panic: runtime error: index out of range [recovered]
	panic: runtime error: index out of range

goroutine 7 [running]:
testing.tRunner.func1(0xc42015c0f0)
	/usr/local/Cellar/go/1.9.2/libexec/src/testing/testing.go:711 +0x2d2
panic(0x434b140, 0x45959f0)
	/usr/local/Cellar/go/1.9.2/libexec/src/runtime/panic.go:491 +0x283
goef/vendor/github.com/spacemonkeygo/openssl.(*encryptionCipherCtx).EncryptUpdate(0xc42000e0b0, 0x45bf568, 0x0, 0x0, 0x20, 0xc4200146b0, 0x10, 0x10, 0x456b9a0)
	/Users/rarayapr/go-workspace/src/goef/vendor/github.com/spacemonkeygo/openssl/ciphers.go:285 +0x21a
goef/pac.(*CryptObject).EncryptData(0xc42005e940, 0x45bf568, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x407ce96)
	/Users/rarayapr/go-workspace/src/goef/pac/cipher.go:61 +0xe0
goef/pac.TestEmptyEncryption(0xc42015c0f0)
	/Users/rarayapr/go-workspace/src/goef/pac/cipher_test.go:98 +0x150
testing.tRunner(0xc42015c0f0, 0x43c45f0)
	/usr/local/Cellar/go/1.9.2/libexec/src/testing/testing.go:746 +0xd0
created by testing.(*T).Run
	/usr/local/Cellar/go/1.9.2/libexec/src/testing/testing.go:789 +0x2de
exit status 2
```

### PR DESCRIPTION
**Pull request by rameshrv**
Hi @zeebo 
 
This PR fixes #95 . Request you to please review.

Thanks,
Ramesh

### CODE REVIEW

**Comment by rameshrv** [comment]
#95

File: ciphers.go [hunk 1]
```diff
@@ -287,6 +287,9 @@ func NewDecryptionCipherCtx(c *Cipher, e *Engine, key, iv []byte) (
 }
 
 func (ctx *encryptionCipherCtx) EncryptUpdate(input []byte) ([]byte, error) {
+	if input == nil {
```
Comment (zeebo) [inline ciphers.go]: I think this actually needs to check for `len(input) == 0`. In other words, passing `[]byte{}` will also panic.
```suggestion
	if len(input) == 0 {
```

I don't actually know what the semantics should be with a zero sized buffer, though. Like, is that even a valid thing one would want to pass?

File: ciphers.go [hunk 2]
```diff
@@ -298,6 +301,9 @@ func (ctx *encryptionCipherCtx) EncryptUpdate(input []byte) ([]byte, error) {
 }
 
 func (ctx *decryptionCipherCtx) DecryptUpdate(input []byte) ([]byte, error) {
+	if input == nil {
```
Comment (zeebo) [inline ciphers.go]: Same here.
```suggestion
	if len(input) == 0 {
```

File: ciphers.go (hunk 1 above)
Comment (rameshrv) [reply ciphers.go]: I believe check for `len(input) == 0` will work for both `nil` and `empty` data. Let me change that and update the PR.

Comment (rameshrv) [reply ciphers.go]: fixed in the new commit. Thanks !

**Pull request merged**